stdlib/Statistics/src/Statistics.jl

```diff
@@ -821,8 +821,11 @@ function quantile!(q::AbstractArray, v::AbstractVector, p::AbstractArray;
     return q
 end
 
-quantile!(v::AbstractVector, p::AbstractArray; sorted::Bool=false) =
-    quantile!(similar(p,float(eltype(v))), v, p; sorted=sorted)
+function quantile!(v::AbstractVector, p::AbstractArray; sorted::Bool=false)
+    T = Core.Compiler.typesubtract(eltype(v), Missing)
+    S = T <: AbstractFloat ? T : promote_type(T, Float64)
```

Comment:
Is there any reason not to use the same approach as method below which is used when `p` is a tuple (which calls `quantilesort!` and then `_quantile`)? `promote_type(T, Float64)` is problematic here, as we claim to support types other than real.

Reply:
Good point. Fixed.